extension/js/popup.js

```diff
@@ -65,6 +73,15 @@
             }
         });
 
+        SiteProfile.retreive(null, function (profiles) {
+            profiles.forEach(function (profile, i) {
+                if (profile.domain === $('#domain').val()) {
+                    siteProfile = profile;
+                    updateDomain();
+                }
+            });
+        });
+
```

Comment:
I used the variable `siteProfile` because `updateDomain` is also called when *Trim domain name* is toggled. I will change the other things you mentioned and leave this change to you.